odoo/orm/table_objects.py

```diff
@@ -188,3 +203,21 @@ def __init__(self, definition: str, message: ConstraintMessageType = ''):
         super().__init__(definition)
         if message:
             self.message = message
+
+
+class TrigramIndex(Index):
+    """ Index on table using pg_trgm extension
+    """
+
+    trigram = True
```

Comment:
You don't need this class, you can just use something like this:
`Index(lambda reg: "USING GIST (t gist_trgm_ops)" if reg.has_trigram else ...)`

Reply:
Yeah but in case we also need to check for `reg.has_unaccent`, we'll need to define the index like
`Index(lambda reg: "USING GIST (unaccent(t) gist_trgm_ops))" if reg.has_trigram and reg.has_unaccent else ("USING GIST (t) gist_trgm_ops" if reg.has_trigram else ""))`

Which is quite verbose so I though a subclass would be neater. And it matches what we are doing when doing `index=trigram` on a field definition.

Also, this means that devs have to think about checking for `reg.has_trigram` as this error is silent (cf. chs comment here: https://github.com/odoo/odoo/pull/166698#discussion_r1677459164 ) and not caught when running CIs on runbot.

That said, if you don't want this subclass, I'll remove it.

Reply:
Alright, I removed the `TrigramIndex` subclass